common/scripted_effects/wc_exodar_effects.txt

```diff
@@ -190,34 +190,49 @@ setup_titles_azuremyst_exodar_effect = {
 		change_county_control = -50
 		add_exodar_crystal_mid_modifier = yes
 	}
-	title:b_valaar.title_province = { add_special_building = exodar_01 }
+	title:b_valaar.title_province = {
+		set_holding_type = city_holding
+		add_special_building = exodar_01 
+	}
 	title:c_yanra = {
 		set_county_culture = culture:draenei
 		set_county_faith = faith:naaru
 		change_development_level = -2
 		change_county_control = -25
 		add_exodar_crystal_low_modifier = yes
+		title_province = {
+			set_holding_type = castle_holding
```

Comment:
I think you said you wanted to make those things through `if` statement?

Reply:
This is true, on the chance the holding is already a castle, then it will throw an error in the error log.